internal/stolonctl/pkg/utils/utils.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright (C) 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gravitational/trace"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/kubelet/kubeletconfig/util/log"
+)
+
+func ConvertError(err error) error {
+	return ConvertErrorWithContext(err, "")
+}
+
+func ConvertErrorWithContext(err error, format string, args ...interface{}) error {
+	if err == nil {
+		return nil
+	}
+	statusErr, ok := err.(*errors.StatusError)
+	if !ok {
+		return err
+	}
+
+	message := fmt.Sprintf("%v", err)
+	if !isEmptyDetails(statusErr.ErrStatus.Details) {
+		message = fmt.Sprintf("%v, details: %v", message, statusErr.ErrStatus.Details)
+	}
+	if format != "" {
+		message = fmt.Sprintf("%v: %v", fmt.Sprintf(format, args...), message)
+	}
+
+	status := statusErr.Status()
+	switch {
+	case status.Code == http.StatusConflict && status.Reason == metav1.StatusReasonAlreadyExists:
+		return trace.AlreadyExists(message)
+	case status.Code == http.StatusNotFound:
+		return trace.NotFound(message)
+	case status.Code == http.StatusForbidden:
+		return trace.AccessDenied(message)
+	}
+	return err
+}
+
+func isEmptyDetails(details *metav1.StatusDetails) bool {
+	if details == nil {
+		return true
+	}
+
+	if details.Name == "" && details.Group == "" && details.Kind == "" && len(details.Causes) == 0 {
+		return true
+	}
+	return false
+}
+
+func Retry(ctx context.Context, times int, period time.Duration, fn func() error) error {
+	if times < 1 {
+		return nil
+	}
+	err := fn()
+	for i := 1; i < times && err != nil; i += 1 {
+		log.Infof("attempt %v, result: %v, retry in %v", i+1, trace.DebugReport(err), period)
```

Comment:
Can you update the logging format to comply with the convention (Capitalized and ending with a dot like a complete sentence)?